enhancements/disaster-recovery/automated-cluster-backups.md

```diff
@@ -0,0 +1,324 @@
+---
+title: automated-cluster-backups
+authors:
+  - "@marun"
+reviewers:
+  - "@hexfusion"
+  - "@lilic"
+  - "@deads2k"
+  - "@smarterclayton"
+approvers:
+  - "@hexfusion"
+  - "@lilic"
+  - "@deads2k"
+  - "@smarterclayton"
+creation-date: 2021-07-08
+last-updated: 2021-07-08
+status: implementable
+see-also:
+  - "https://docs.google.com/document/d/1LyZQMgLvgd81iQndcDcjAVN25ZEhCxVGMttoiRkSU6g/edit#"
+  - "https://docs.openshift.com/container-platform/4.7/backup_and_restore/backing-up-etcd.html"
+  - "https://rancher.com/docs/rancher/v2.x/en/backups/v2.5/configuration/backup-config/"
+  - "https://www.openshift.com/blog/ocp-disaster-recovery-part-1-how-to-create-automated-etcd-backup-in-openshift-4.x"
+---
+
+# Automated Cluster Backups
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Enable the configuration of automated cluster backups.
+
+## Motivation
+
+The current documented procedure for performing a backup of an OpenShift
+cluster is manually initiated. This reduces the likelyhood of a timely backup
+being available in the event of catastrophic cluster failure.
+
+The procedure also requires gaining a root shell on a control plane
+node. Shell access to OpenShift control plane nodes access is generally
+discouraged due to the potential for affecting the reliability of the
+node. Indeed, at least one customer reports having had operator error result
+in a control plane node being taken offline in the process of attempting to
+follow the documented backup procedure.
+
+Finally, OpenShift 4.9 is intended to ship with etcd 3.5, and this will mean
+an upgrade for existing clusters from etcd 3.4. Since etcd will not support
+downgrade from 3.5 to 3.4, downgrade of a OpenShift cluster from 4.9 to 4.8
```

Comment:
If your desire in 4.9 is to irreversibly switch to etcd 3.5, then this feature needs to save a backup *before* upgrading to 4.9, which suggests you would need this feature in 4.8.  

1. Is the timing listed here stale (and you mean 4.10)?
2. Is this supposed to be backportable?
3. Is 4.9 going to have two stages of upgrade that involve "updated to 3.4+backup, then updated to 3.5"?
4. Is 4.8 going to have a manual, one-off backup and then status reporting?

Reply:
@hexfusion can confirm, but https://github.com/openshift/enhancements/pull/830#discussion_r671425446 suggests a desire to block upgrades to 4.9 unless a recent backup exists.

Reply:
Yes, I agree we might have to backport something to 4.8 to ensure a backup is taken. From my small experiments, it wasn't possible to easily downgrade due to the schema changes.